### PR TITLE
Add limited spot support to ADAM überscript (resolves #256)

### LINKED ISSUE #256: Add spot pricing back to ADAM überscript
We removed support for spot instances in https://github.com/BD2KGenomics/toil-scripts/commit/2d9a0abfba0e6eb21388dc56cfdfe07f056649a9#diff-06a900fff8217905f65f50fa4ee0e9a9. We should add this back, and fix various syntax issues in the überscript.

### PR DESCRIPTION
**Pull request by fnothaft**
* Returns spot instance support to the ADAM überscript. If a `--spot-price` option is passed on the command line, calls to `cgcloud` to start/grow a toil cluster will pass a spot price. If this option is not passed, on demand instances will be used.
* Fixes a syntactical issue with the `grow_cluster` method in the ADAM überscript that led to the optional `cluster_type` being incorrectly overridden when trying to pass the `*options` parameter.

### CODE REVIEW

File: src/toil_scripts/adam_uberscript/adam_uberscript.py [hunk 1]
```diff
@@ -262,10 +263,13 @@ def update_cluster_size(domain, n):
         cluster_size.save()
 
 
-def grow_cluster(num_nodes, instance_type, cluster_name, cluster_type='toil', *options):
+def grow_cluster(num_nodes, instance_type, cluster_name, cluster_type = 'toil', options = None):
     """
     Grow a cluster by a given number of nodes
     """
+    if not options:
```
Comment (hannes-ucsc) [inline src/toil_scripts/adam_uberscript/adam_uberscript.py]: `if options is None`

File: src/toil_scripts/adam_uberscript/adam_uberscript.py [hunk 2]
```diff
@@ -262,10 +263,13 @@ def update_cluster_size(domain, n):
         cluster_size.save()
 
 
-def grow_cluster(num_nodes, instance_type, cluster_name, cluster_type='toil', *options):
+def grow_cluster(num_nodes, instance_type, cluster_name, cluster_type = 'toil', options = None):
```
Comment (hannes-ucsc) [inline src/toil_scripts/adam_uberscript/adam_uberscript.py]: I'm pretty sure PEP8 disallows spaces around = in kwargs.

File: src/toil_scripts/adam_uberscript/adam_uberscript.py [hunk 3]
```diff
@@ -345,7 +358,7 @@ def manage_toil_cluster(params, domain):
                              cluster_size,
                              desired_cluster_size)
                     num_nodes = desired_cluster_size - cluster_size
-                    grow_cluster(num_nodes, params.instance_type, params.cluster_name, *role_options(params))
+                    grow_cluster(num_nodes, params.instance_type, params.cluster_name, options = (role_options(params) + spot_options(params)))
```
Comment (hannes-ucsc) [inline src/toil_scripts/adam_uberscript/adam_uberscript.py]: See above re spaces. An parentheses are redundant here.

**Comment by hannes-ucsc** [comment]
Only nits.

**Comment by fnothaft** [comment]
Nits addressed.

**Comment by hannes-ucsc** [comment]
@hannes_ucsc, merge once tests pass.

**Pull request merged**